pkg/tests/api/alerting/api_notification_channel_test.go

```diff
@@ -30,6 +30,192 @@ import (
 	"github.com/grafana/grafana/pkg/tests/testinfra"
 )
 
+func TestTestReceivers(t *testing.T) {
+	t.Run("assert working receiver returns OK", func(t *testing.T) {
+		// Setup Grafana and its Database
+		dir, path := testinfra.CreateGrafDir(t, testinfra.GrafanaOpts{
+			EnableFeatureToggles: []string{"ngalert"},
+		})
+		store := testinfra.SetUpDatabase(t, dir)
+		store.Bus = bus.GetBus()
+		grafanaListedAddr := testinfra.StartGrafana(t, dir, path, store)
+		require.NoError(t, createUser(t,
+			store,
+			models.ROLE_EDITOR,
+			"grafana",
+			"password"))
+
+		oldEmailBus := bus.GetHandlerCtx("SendEmailCommandSync")
+		mockEmails := &mockEmailHandler{}
+		bus.AddHandlerCtx("", mockEmails.sendEmailCommandHandlerSync)
+		t.Cleanup(func() {
+			bus.AddHandlerCtx("", oldEmailBus)
+		})
+
+		testReceiversURL := fmt.Sprintf("http://grafana:password@%s/api/alertmanager/grafana/config/api/v1/receivers/test", grafanaListedAddr)
+		// nolint:gosec
+		resp, err := http.Post(testReceiversURL, "application/json", strings.NewReader(`{
+	"receivers": [{
+		"name":"receiver-1",
+		"grafana_managed_receiver_configs": [
```

Comment:
how do you feel about expanding this example (or the one where we timeout) to include a mixture of failure + success to make sure we exercise that path?